lib/cdo/poste.rb

```diff
@@ -170,12 +171,11 @@ def self.ensure_recipient(email, params={})
     ip_address = params[:ip_address]
     now = DateTime.now
 
-    contacts = POSTE_DB[:contacts]
-
-    contact = contacts.where(hashed_email: hashed_email).first
+    contact = POSTE_DB[:contacts].where(hashed_email: hashed_email).first
     unless contact
-      id = contacts.insert({}.tap do |contact|
-        contact[:email] = email
+      sanitized_email = Poste.dashboard_student?(hashed_email) ? '' : email
+      id = POSTE_DB[:contacts].insert({}.tap do |contact|
+        contact[:email] = sanitized_email
```

Comment:
Not important: I know you didn't write this code, and it's mostly extraneous to this PR, but any idea why it's using `{}.tap` rather than just creating a hash inline? 

That seems a strange and unnecessary pattern. The only reason I can see is the optional name field, but since this is inserting in the DB wouldn't a hash with `name: nil` resolve to the same thing as as a hash with no `name` key?

It can be converted to:

```ruby
id = POSTE_DB[:contacts].insert({
  email: sanitized_email,
  hashed_email: hashed_email,
  name: name,
  #...
})
```